common/metrics/opentelemetry_provider.go

```diff
@@ -102,7 +103,12 @@ func NewOpenTelemetryProvider(
 	return reporter, nil
 }
 
-func initPrometheusListener(config *PrometheusConfig, reg *prometheus.Registry, logger log.Logger) *http.Server {
+func initPrometheusListener(
+	config *PrometheusConfig,
+	reg *prometheus.Registry,
+	logger log.Logger,
+	onError func(e error),
```

Comment:
maybe call it onListenError to be more specific
```suggestion
	onListenError func(error),
```

Reply:
I'd also be fine with just a `listenErrorIsFatal bool` :shrug:

Reply:
makes sense. switched to bool flag.